rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,152 @@
+- `Example.tsx` - (complete component to be used as is)
+- `useExample.ts` - (CONTROLLER/MODEL /main logic behind component - re-usable hook)
+- `useExampleStyles.ts` - (STYLES/ main logic for styling component - re-usable hook)
+- `useExampleRandom.tsx` - (some encapsulated logic - not really a hook)
+- `renderExample.tsx` - (VIEW/ main logic that handles rendering - re-usable hook)
+- `Example.types.ts` - (TYPES/ all TypeScript types used within package)
+
+**Potential issues from developer point of view:**
+
+- inconsistent file naming
+  - some files contain suffix (`.types`)
+  - some files have `.ts` some `.tsx` what's the difference ?
+  - some files are PascalCased, some camelCased
+- not explicitly communicating the intent
+  - what does `use` mean ?
+    - seasoned react dev will probably guess that it's a hook. Thing is, that this file might export more that 1 hook.
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+**Proposed rules applied on current state:**
+
+> see [Explanation](#Explanation) for more details about applied rules
+
+| Current File Name   | Proposed File name  | Symbol Name                                                                         | Remarks                                                                                                                                                                                                                                                                                                                   |
+| ------------------- | ------------------- | ----------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| index.tsx           | index.ts            |                                                                                     | No changes from current                                                                                                                                                                                                                                                                                                   |
+| Example.tsx         | example.ts          | `export const Example = () => {..}`                                                 | To enforce rendering logic only within `.view.` type file/s. TS will enforce this by default. Not extremely picky on this one, we can use `.tsx` as well to match extension with test/story - priority is to be consistent                                                                                                |
```

Comment:
```suggestion
| Example.tsx         | Example.ts          | `export const Example = () => {..}`  |
```

- it's a common pattern to use Pascal Case for components: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md
- it's a common pattern to match filename to default export:
  - https://github.com/mui-org/material-ui/tree/next/packages/material-ui/src/Accordion
  - https://github.com/adobe/react-spectrum/tree/main/packages/%40react-spectrum/alert/src
  - https://github.com/reakit/reakit/tree/master/packages/reakit/src/Box
  - https://github.com/Semantic-Org/Semantic-UI-React/tree/master/src/elements/Button
  - https://github.com/segmentio/evergreen/tree/master/src/avatar/src
- React team uses Pascal Case in docs https://reactjs.org/docs/code-splitting.html#reactlazy

Reply:
I provided explicit reasoning in this document. This kind of feedback is not very helpful TBH.

- we are not react team
- default exports are an anti-pattern (nor is fluent using that ECMA feature)
- we can start passing links indefinitely, to showcase who uses what :)

Reply:
I agree with Shift on this one, I'd prefer to keep the uppercase filename basically for the reasons he listed. 
- Clearly indicates that this file exports a component (even if it doesn't contain any rendering logic)
- Matches the name of the main/only export (not relevant for this purpose whether that export is default or named)
- Matches the convention used by both v8 and v0 today, and is extremely common throughout the community. Even though nobody has declared this as the "official" convention, at least from briefly looking through other React component repos, it's harder to find a repo that *doesn't* use uppercase names for component files than one that does.